src/lib/utils/check_table_equal.cpp

```diff
@@ -95,6 +95,33 @@ bool almost_equals(T left_val, T right_val, FloatComparisonMode float_comparison
 }  // namespace
 
 namespace opossum {
+
+bool check_segment_equal(const std::shared_ptr<BaseSegment>& segment_to_test,
+                         const std::shared_ptr<BaseSegment>& expected_segment, OrderSensitivity order_sensitivity,
+                         TypeCmpMode type_cmp_mode, FloatComparisonMode float_comparison_mode) {
+  if (segment_to_test->data_type() != expected_segment->data_type()) {
+    return false;
+  }
+
+  const auto definitions =
+      std::vector<TableColumnDefinition>{TableColumnDefinition("single_column", segment_to_test->data_type(), true)};
+
+  auto table_type = [&](const std::shared_ptr<BaseSegment> segment) {
+    if (const auto reference_segment = std::dynamic_pointer_cast<const ReferenceSegment>(segment)) {
+      return TableType::References;
+    }
+    return TableType::Data;
+  };
+
+  auto table_to_test = std::make_shared<Table>(definitions, table_type(segment_to_test));
+  table_to_test->append_chunk(pmr_vector<std::shared_ptr<BaseSegment>>{segment_to_test});
+  auto expected_table = std::make_shared<Table>(definitions, table_type(expected_segment));
+  expected_table->append_chunk(pmr_vector<std::shared_ptr<BaseSegment>>{expected_segment});
+
+  // The optional returned by check_table_equal is set when the function failed to determine equality.
```

Comment:
Took me three times to understand. Maybe “If `check_table_equal` returns something other than std::nullopt, a difference has been found.”